src/main/java/com/relayrides/pushy/apns/PushManager.java

```diff
@@ -824,13 +824,13 @@ public void handleUnprocessedNotifications(ApnsConnection<T> connection, Collect
 	}
 
 	private void startNewConnection() {
-		synchronized (this.activeConnections) {
-			final ApnsConnection<T> connection = new ApnsConnection<T>(this.environment, this.sslContext,
-					this.eventLoopGroup, this.configuration.getConnectionConfiguration(), this,
-					String.format("%s-connection-%d", this.name, this.connectionCounter++));
+		final ApnsConnection<T> connection = new ApnsConnection<T>(this.environment, this.sslContext,
+				this.eventLoopGroup, this.configuration.getConnectionConfiguration(), this,
+				String.format("%s-connection-%d", this.name, this.connectionCounter++));
```

Comment:
`++` is not thread-safe, so `connectionCounter` would need to be changed to an `AtomicInteger`.